horizons/gui/widgets/tooltip.py

```diff
@@ -97,103 +101,98 @@ def position_tooltip(self, event):
-		self.bg.amount = len(tooltip.splitlines()) - 1 + bool(buildmenu_icons)
-		self.label.text = bool(buildmenu_icons) * '\n' + tooltip
-		self.gui.adaptLayout()
-		self.gui.show()
+		else:
+			self.wrap_tt_text(buildmenu_icons, tooltip_text)
 
-		# NOTE: the below code in this method is a hack to resolve #2227
-		# cannot find a better way to fix it, cause in fife.pychan, it seems
-		# if a widget gets hidden or removed, the children of that widget are not
-		# hidden or removed properly (at least in Python code)
-
-		# update topmost_widget every time the tooltip is shown
-		# this is to dismiss the tooltip later, see _check_hover_alive
-		target_widget = self
-		while target_widget:
-			self.topmost_widget = target_widget
-			target_widget = target_widget.parent
-
-		# add an event to constantly check whether the hovered widget is still there
-		# if this is no longer there, dismiss the tooltip widget
-		ExtScheduler().add_new_object(self._check_hover_alive, self, run_in=0.5, loops=-1)
-
-	def _check_hover_alive(self):
-		target_widget = self
-		# traverse the widget chain again
-		while target_widget:
-			# none of ancestors of this widget gets removed, 
-			# just do nothing and let the tooltip shown
-			if target_widget == self.topmost_widget:
-				return
-			# one ancestor of this widget is hidden
-			if not target_widget.isVisible():
-				self.hide_tooltip()
-				return
-			target_widget = target_widget.parent
+		# For large tooltips the width of the label is off because we are not using a monospaced font
+		# so the margins have to be adapted. Since the label is already initialized, we have to check whether the
+		# tooltip is big, adapt the margins, hide and reshow the tooltip.
+		if self.label.width >= 100:
+			self.LABEL_LEFT_MARGIN = 12
+			self.LABEL_RIGHT_MARGIN = 12
+			self.label.position = self.LABEL_LEFT_MARGIN, self.LABEL_TOP_BOTTOM_MARGIN
 
-		# if it comes to here, meaning one ancestor of this widget is removed
-		self.hide_tooltip()
+		# self.bg.amount should be renamed to self.bg.y_tile_amount or something similar.
+		# unfortunately there are several xml files that use amount and it would be a lot of work to fix this
+		self.bg.amount = ((self.label.height / 18) - 1)
+		# set the width of the icon
+		self.bg.x_width_amount = (self.MAX_BG_WIDTH + self.LABEL_LEFT_MARGIN + self.LABEL_RIGHT_MARGIN)
+
+		HoverInstancesChanged.subscribe(self.hide_tooltip)
 
 	def hide_tooltip(self, event=None):
 		if self.gui is not None:
+			self.tooltip_shown = False
 			self.gui.hide()
-		# tooltip is hidden, no need to check any more
-		ExtScheduler().rem_call(self, self._check_hover_alive)
-		self.topmost_widget = None
-		self.tooltip_shown = False
+
+	def wrap_tt_text(self, buildmenu_icons, text):
+		chars_per_line = 1
+		for i in range(0, text.__len__()):
+
+			if self.label.text.__sizeof__() < self.MAX_LABEL_WIDTH:
+				self.label.text += text[i]
+				chars_per_line += 1
+			self.gui.adaptLayout()
+		chars_per_line - 1
```

Comment:
This line does nothing at all O.o Maybe there's an `=` missing? But then how could it be needed, if everything worked like this?